deploy.go

```diff
@@ -188,24 +188,33 @@ func (d *imageDeployer) computePCRProtectionProfile(esp string, env secboot_efi.
 	log.Infoln("computing PCR protection profile")
 	pcrProfile := secboot_tpm2.NewPCRProtectionProfile()
 
-	loadSequences := []*secboot_efi.ImageLoadEvent{
-		{
-			Source: secboot_efi.Firmware,
-			Image:  secboot_efi.FileImage(filepath.Join(esp, "EFI/ubuntu/shimx64.efi")),
-			Next: []*secboot_efi.ImageLoadEvent{
-				{
-					Source: secboot_efi.Shim,
-					Image:  secboot_efi.FileImage(filepath.Join(esp, "EFI/ubuntu/grubx64.efi")),
-				},
-			},
-		},
+	// This function assumes that the boot architecture is Azure FDE (no grub)
+	kernelPaths, err := filepath.Glob(filepath.Join(esp, "EFI/ubuntu/kernel.efi-*"))
+	if err != nil {
+		return nil, xerrors.Errorf("cannot determine kernel paths: %w", err)
+	}
+	if _, err := os.Stat(filepath.Join(esp, "EFI/ubuntu/grubx64.efi")); err == nil {
+		kernelPaths = append(kernelPaths, filepath.Join(esp, "EFI/ubuntu/grubx64.efi"))
 	}
```

Comment:
This is really obscure. Maybe explaining that the kernel was installed here by earlier version of the kernel package would help understand this code.